### DISCUSSION TITLE
Short lived near caches at scale a good or bad idea?

### DISCUSSION THREAD

**Discussion by javafanboy** [oracle/coherence#80] [Q&A] [answered]
Is there anything in the Coherence architecture or implementation that makes it a bad idea to have relatively short lived near caches (I am here talking about life times from perhaps tens of minutes to say a day)?
I have measured the time it takes to create new caches at small scale and it seems quite ok but I am here thinking on how it would work at larger scale...
I.e. if such caches are created and destroyed at scale (hundreds or perhaps thousands per day) would create a very high load on the cluster holding the backend distributed caches taking GC and other factors into consideration and could there be are internal resources that could eventually "run out" (like some cache ids etc)?

**Comment by aseovic** [comment, marked as answer]
The cost on the storage members will be limited to the cost to register and unregister event listeners used by the near cache, so it shouldn't be significant. Of course, there is also the cost to dispatch invalidation events based on the chosen invalidation strategy, but that has more to do with how many near caches you have at any given time, not so much with how long they are around for.

Hundreds should be fine, thousands _may_ be ok, 100s of thousands or more is probably too many ;-)
